Birthdays: Not posting birthdays

Hey, so added a couple debug statements within the code as I've noticed birthdays havent posted and I noticed

```py
Oct 06 13:47:25 flare python[19674]: [2020-10-06 13:47:25] [ERROR] red.ZeCogsV3.birthdays: aaaa
Oct 06 13:47:25 flare python[19674]: Traceback (most recent call last):
Oct 06 13:47:25 flare python[19674]:   File "/home/flare/bots/editedcogs/birthdays/birthdays.py", line 71, in initialise

Oct 06 13:47:25 flare python[19674]:     await self.clean_yesterday_bdays()
Oct 06 13:47:25 flare python[19674]:   File "/home/flare/bots/editedcogs/birthdays/birthdays.py", line 259, in clean_yesterday_bdays
Oct 06 13:47:25 flare python[19674]:     await self.config.guild(discord.Guild(data={"id": guild_id}, state=None)).yesterdays.clear()
Oct 06 13:47:25 flare python[19674]:   File "/home/flare/redenv/lib/python3.8/site-packages/discord/guild.py", line 190, in __init__
Oct 06 13:47:25 flare python[19674]:     self._from_data(data)
Oct 06 13:47:25 flare python[19674]:   File "/home/flare/redenv/lib/python3.8/site-packages/discord/guild.py", line 308, in _from_data
Oct 06 13:47:25 flare python[19674]:     cache_online_members = self._state._member_cache_flags.online
Oct 06 13:47:25 flare python[19674]: AttributeError: 'NoneType' object has no attribute '_member_cache_flags'```

Gonna try have a look and see if I can solve it once I get some spare time

--- a/birthdays/birthdays.py
+++ b/birthdays/birthdays.py
@@ -66,8 +66,14 @@ class Birthdays(Cog):
                 now = datetime.datetime.utcnow()
                 tomorrow = (now + datetime.timedelta(days=1)).replace(hour=0, minute=0, second=0, microsecond=0)
                 await asyncio.sleep((tomorrow - now).total_seconds())
-                await self.clean_yesterday_bdays()
-                await self.do_today_bdays()
+                try:
+                    await self.clean_yesterday_bdays()
+                except Exception as e:
+                    self.logger.error("An error occured during cleaning up of previous birthdays.", exc_info=e)
+                try:
+                    await self.do_today_bdays()
+                except Exception as e:
+                    self.logger.error("An error occured during doing todays birthdays.", exc_info=e)
 
     def cog_unload(self):
         self.bday_loop.cancel()  # Forcefully cancel the loop when unloaded
@@ -220,7 +226,7 @@ class Birthdays(Cog):
         for guild_id, guild_config in all_guild_configs.items():
             for user_id in guild_config.get("yesterdays", []):
                 asyncio.ensure_future(self.clean_bday(guild_id, guild_config, user_id))
-            await self.config.guild(discord.Guild(data={"id": guild_id}, state=None)).yesterdays.clear()
+            await self.config.guild_from_id(guild_id).yesterdays.clear()
 
     async def do_today_bdays(self):
         bday_configs = await self.get_all_date_configs()

